trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/index.ts

```diff
@@ -22,16 +24,25 @@ const renderUtilisationReportReconciliationForReport = (res: Response, viewModel
 export const getUtilisationReportReconciliationByReportId = async (req: Request, res: Response) => {
   const { userToken, user } = asUserSession(req.session);
   const { reportId } = req.params;
-  const { facilityIdQuery } = req.query;
+  const { facilityIdQuery, selectedFeeRecordIds: selectedFeeRecordIdsQuery } = req.query;
 
   try {
-    const facilityIdQueryAsString = facilityIdQuery ? asString(facilityIdQuery, 'facilityIdQuery') : undefined;
-    const facilityIdQueryError = validateFacilityIdQuery(facilityIdQueryAsString, req.originalUrl);
-    const { errorSummary: premiumPaymentFormError, isCheckboxChecked } = getAndClearFieldsFromRedirectSessionData(req);
+    const facilityIdQueryString = facilityIdQuery ? asString(facilityIdQuery, 'facilityIdQuery') : undefined;
+
+    const filterError = validateFacilityIdQuery(facilityIdQueryString, req.originalUrl);
+
+    const { tableError, selectedFeeRecordIds: selectedFeeRecordIdsFromSessionData } = getAndClearFieldsFromRedirectSessionData(req);
+
+    const selectedFeeRecordIdsQueryString = selectedFeeRecordIdsQuery ? asString(selectedFeeRecordIdsQuery, 'selectedFeeRecordIdsQuery') : undefined;
+
+    const selectedFeeRecordIds: Set<number> =
+      selectedFeeRecordIdsFromSessionData.size > 0 ? selectedFeeRecordIdsFromSessionData : getSelectedFeeRecordIdsFromQuery(selectedFeeRecordIdsQueryString);
```

Comment:
Should we move some of this logic into its own function/helper, think it would be easier to digest, thinking something like this - what do you think?

```ts
const queryAndSessionThings = newFunction();

// ... await api.getUtilisationReportReconciliationDetailsById(

const formattedReportPeriod = getFormattedReportPeriodWithLongMonth(reportPeriod);

const enablePaymentsReceivedSorting = feeRecordPaymentGroupsHaveAtLeastOnePaymentReceived(feeRecordPaymentGroups);

const feeRecordPaymentGroupViewModel = mapFeeRecordPaymentGroupsToFeeRecordPaymentGroupViewModelItems(feeRecordPaymentGroups, isCheckboxChecked);

const keyingSheetViewModel = mapKeyingSheetToKeyingSheetViewModel(keyingSheet);

const paymentDetailsViewModel = mapFeeRecordPaymentGroupsToPaymentDetailsViewModel(feeRecordPaymentGroups);

return renderUtilisationReportReconciliationForReport(
    // ...
)
```

Reply:
I think this could be nice, but I think this would add a fair amount more work to the PR as the new helper will have to have additional tests written.

I feel like it's fairly understandable from the spacing and documentation changes you mentioned in previous comments.

I'm fairly keen to try and get this PR merged as the sprint's ending tomorrow and there's been a fair amount of additional changes already, although if you're very keen for this to be pulled out I'm happy to give it a go. Thanks

Reply:
Thanks @sw-mattw understood, but I think we should move the new logic into a new helper, since it's things that have been introduced in this PR.

I think otherwise the controller feels pretty noisy as there's a lot going on. I worry if we don't move these now, it won't get moved. Looks like it would only take 10-15mins including tests? We could pair or I could pick this up if you'd like 

Apart from this and 2/3 other minor comments, PR looks great, nice work 😊

Reply:
Fair point, pulled this out to a new function `extractQueryAndSessionData` if you wouldn't mind reviewing that please in:

`trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/extract-query-and-session-data.ts`